File: problems/3005/solution.erl
-spec max_frequency_elements([integer()]) -> integer().
max_frequency_elements(Nums) ->
    Freq = lists:foldl(fun(X, M) -> maps:update_with(X, fun(V) -> V + 1 end, 1, M) end, #{}, Nums),
    Values = maps:values(Freq),
    Best = lists:max(Values),
    lists:sum([V || V <- Values, V =:= Best]).

File: problems/3005/solution.rb
# @param {Integer[]} nums
# @return {Integer}
def max_frequency_elements(nums)
  freq = Hash.new(0)
  nums.each { |x| freq[x] += 1 }
  best = freq.values.max
  freq.values.select { |count| count == best }.sum
end
